### PR TITLE
PyTorch nn Layers entry

### PR DESCRIPTION
**Pull request by DigitalDruid10010110**
### Description

This PR includes the completed draft of the 'Layers' documentation. The document provides an explanation of layers in PyTorch, syntax examples using `nn.Sequential()`, and a simple multi-layer model.  My motivation for this project comes from my portfolio project for "Learn Git & GitHub" course on Codecademy.  

This was a difficult project for me as I had no previous experience in machine learning or PyTorch let alone knowledge of layers.  The process of learning and taking the courses to learn enough to create this document was really rewarding and helped break down some walls of fear and anxiety.  I look forward to your review and any suggestions you have.

### Issue Solved

 [Term Entry] PyTorch nn Layers #5230 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+Title: 'layers'
+Description: 'Layers and their role in PyTorch neural networks.'
```
Comment (PragatiVerma18) [inline content/pytorch/concepts/nn/terms/layers/layers.md:3]: The description should tell what layers are and it should be repeated in the introductory paragraph of the entry as per the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md). 

We can define it as following -

```suggestion
Description: 'Layers are the fundamental building blocks used to construct neural networks.' 
```
or 

```suggestion
Description: 'Layers are modules that perform operations on input data to build neural networks.'
```

Please update the same in the first paragraph based on your choice.

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - 'AI'
+  - 'Deep Learning'
+  - 'Models'
+CatalogContent:
+  - 'learn-intro-to-py-torch-and-neural-networks'
+  - 'paths/machine-learning'
+---
+
+**Layers** are higher-level constructs built from **Tensors**, used to define the architecture of neural networks in PyTorch. Neural networks are created by stacking layers in specific patterns to transform input data into outputs, such as image classifications or predictions. PyTorch offers various layers, including **Linear**, **Convolutional**, **Recurrent**, **Pooling**, and **Transformer layers**, each optimized for tasks like image recognition or natural language processing. By stacking multiple layers, neural networks can learn complex patterns, making them effective for a variety of AI applications.
+
```
[the remark this answers is not in the document]
Comment (PragatiVerma18) [reply content/pytorch/concepts/nn/terms/layers/layers.md:19]: We should also skip the names of different layers for simplicity here as the reader is still trying to understand what are layers as they are reading and this is a step ahead.

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Syntax
+
+By using PyTorch's sequential() method to build a neural network, we can specify **layers** and activation functions in sequence from input to output as show below:
```
Comment (PragatiVerma18) [inline content/pytorch/concepts/nn/terms/layers/layers.md:22]: When referring to properties and methods, we should mention the exact function name along with `.` and `()` to help the user relate the text to the actual syntax.

Also, bold should only be used when the term is introduced for the first time. 

The ending `n` is missing in `shown`.

```suggestion
By using PyTorch's `.Sequential()` method to build a neural network, we can specify layers and activation functions in sequence from input to output as shown below:
```

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import torch
+import torch.nn as nn
+
+# Define the model for the neural network
+model = nn.Sequential(
+    # Define layers and activation functions here
+    nn.Linear(in_features, out_features)
+    nn.ReLU()
+    nn.Linear(in_features, out_features)
+)
```
Comment (PragatiVerma18) [inline content/pytorch/concepts/nn/terms/layers/layers.md:34]: In `nn.Sequential()` method, each layer should be separated by a comma. 

Here’s the corrected version:

```suggestion
model = nn.Sequential(
    # Define layers and activation functions here
    nn.Linear(in_features, out_features),
    nn.ReLU(),
    nn.Linear(in_features, out_features)
)
```

The commas are essential in Python to separate items within argument lists like `nn.Sequential()`.

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+## Example
+
+Below is an example of a basic Multi-**Layer** sequential neural network where data flows down through the specified sequence of layers and activation functions:
```
Comment (PragatiVerma18) [inline content/pytorch/concepts/nn/terms/layers/layers.md:39]: The term 'Layer' should be highlighted in bold only when it is used for the first time. 

The term "Multi-Layer" should be written as "multi-layer" (lowercase and hyphenated) as it is not a part of a title.

```suggestion
Below is an example of a basic multi-layer sequential neural network where data flows down through the specified sequence of layers and activation functions:
```

Comment (PragatiVerma18) [reply content/pytorch/concepts/nn/terms/layers/layers.md:39]: The phrase "data flows down" could be rephrased to use active voice for clarity. For example, "where the data is processed in a top-down manner" could make the action clearer.

Here is the complete sentence for reference - 

```suggestion
Below is an example of a basic multi-layer neural network defined using PyTorch's `nn.Sequential()` method, where the data is processed through a specified sequence of layers and activation functions:
```

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import torch.nn as nn
+
+model = nn.Sequential(
+    nn.Linear(56,128),
+    nn.ReLU(),
+    nn.Linear(128,26),
+    nn.ReLU(),
+    nn.Linear(26,1)
+)
+```
```
Comment (PragatiVerma18) [inline content/pytorch/concepts/nn/terms/layers/layers.md:52]: I would suggest adding relevant code comments to tell the user what is happening in different steps.

```suggestion
```py
import torch
import torch.nn as nn

model = nn.Sequential(
    # First layer: Linear transformation from 56 input features to 128 output features
    nn.Linear(56, 128),  # This layer learns to transform input data
    nn.ReLU(),           # Activation function to introduce non-linearity
    
    # Second layer: Linear transformation from 128 input features to 26 output features
    nn.Linear(128, 26),  # This layer continues to transform the data
    nn.ReLU(),           # Another activation function for non-linearity
    
    # Third layer: Linear transformation from 26 input features to 1 output feature
    nn.Linear(26, 1)     # Final output layer that produces the model's prediction
)
```
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  - 'Machine Learning'
+Tags:
+  - 'Machine Learning'
+  - 'Neural Networks'
+  - 'AI'
+  - 'Deep Learning'
+  - 'Models'
+CatalogContent:
+  - 'learn-intro-to-py-torch-and-neural-networks'
+  - 'paths/machine-learning'
```
[the remark this answers is not in the document]
Comment (DigitalDruid10010110) [reply content/pytorch/concepts/nn/terms/layers/layers.md:15]: Thank you for the resource.  I will get this updated.

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+**Layers** are higher-level constructs built from **Tensors**, used to define the architecture of neural networks in PyTorch. Neural networks are created by stacking layers in specific patterns to transform input data into outputs, such as image classifications or predictions. PyTorch offers various layers, including **Linear**, **Convolutional**, **Recurrent**, **Pooling**, and **Transformer layers**, each optimized for tasks like image recognition or natural language processing. By stacking multiple layers, neural networks can learn complex patterns, making them effective for a variety of AI applications.
```
[the remark this answers is not in the document]
Comment (DigitalDruid10010110) [reply content/pytorch/concepts/nn/terms/layers/layers.md]: Updated with suggestion 2.

File: content/pytorch/concepts/nn/terms/layers/layers.md (hunk 1 above)
Comment (DigitalDruid10010110) [reply content/pytorch/concepts/nn/terms/layers/layers.md]: Updated description with option 2 to reflect new opening paragraph.

File: content/pytorch/concepts/nn/terms/layers/layers.md (hunk 3 above)
Comment (DigitalDruid10010110) [reply content/pytorch/concepts/nn/terms/layers/layers.md]: Updated to suggested change.

**Comment by DigitalDruid10010110** [comment]
Hello @PragatiVerma18, I had made a mistake on a commit comment and had to update it and force push it to update my commit history to properly reflect the changes made during the commit that I made.  There were no conflicts, but it did remove the verified tag on commits that came after the one I had to update.  I apologize if this has made more work for you.  Thank you for you patience.

**Review by PragatiVerma18 — APPROVED** [review]
Hey @DigitalDruid10010110, the changes look good to me, I am marking it ready for the next review. Thanks for the contribution.

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+Description: 'Layers are modules that perform operations on input data to build neural networks.' 
+Subjects:
+  - 'AI'
+  - 'Machine Learning'
+Tags:
+  - 'Machine Learning'
+  - 'Neural Networks'
+  - 'AI'
+  - 'Deep Learning'
+  - 'Models'
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/layers/layers.md:12]: The tags should be in alphabetical order.
```suggestion
  - 'AI'
  - 'Deep Learning'
  - 'Machine Learning'
  - 'Models'
  - 'Neural Networks'
```

**Comment by Radhika-okhade** [comment]
Hey! @DigitalDruid10010110 The entry looks good. I have made a few minor suggestions. Could you please look into them?

File: content/pytorch/concepts/nn/terms/layers/layers.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  - 'Machine Learning'
+  - 'Models'
+  - 'Neural Networks'
+CatalogContent:
+  - 'learn-intro-to-py-torch-and-neural-networks'
+  - 'py-torch-for-classification'
+  - 'paths/machine-learning'
+---
+
+Layers are modules that perform operations on input data to build neural networks.
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/nn/terms/layers/layers.md:19]:
```suggestion
**Layers** are modules that perform operations on input data to build neural networks.
```

**Review by Radhika-okhade — APPROVED** [review]

**Review by Radhika-okhade — APPROVED** [review]

**Review by Radhika-okhade — APPROVED** [review]

**Pull request merged**